### PR TITLE
Ask for your name when generating a new keyboard

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Currently the keyboard generator script decides what your name is for you. Now it will ask :^)

`REPLACE_WITH_YOUR_NAME` has been changed to `%YOUR_NAME%` and will be substituted in:
 - config.h (Copyright header and `MANUFACTURER`)
 - readme.md (Maintainer link text, but not GitHub username)
 - *keyboard*.c (Copyright header)
 - *keyboard*.h (Copyright header)
 - keymaps/default/config.h (Copyright header)
 - keymaps/default/keymap.c (Copyright header)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
#5191 would make this obsolete, correct?

**Comment by fauxpark** [comment]
Indeed. This can be closed if you're aiming to merge that instead.

File: util/new_project.sh [hunk 1]
```diff
@@ -43,7 +43,11 @@ GIT=$(whereis git)
 if [ "$GIT" != "" ]; then
   IS_GIT_REPO=$($GIT log >>/dev/null 2>&1; echo $?)
   if [ "$IS_GIT_REPO" -eq 0 ]; then
-    ID="'$($GIT config --get user.name)'"
+    ID="$($GIT config --get user.name)"
+    read -p "What is your name? [$ID] " YOUR_NAME
```
Comment (mechmerlin) [inline util/new_project.sh]:
```suggestion
    read -rp "What is your name? [$ID] " YOUR_NAME
```

should put a -r there in the strange event that someone wants to use backslashes for their name lol.

File: util/new_project.sh [hunk 2]
```diff
@@ -43,7 +43,11 @@ GIT=$(whereis git)
...
+    if [ ! -z "$YOUR_NAME" ]; then
```
Comment (mechmerlin) [inline util/new_project.sh]:
```suggestion
    if [ -n "$YOUR_NAME" ]; then
```

`-n` would work better here than a `! -z`

**Pull request merged**